types/artifacts.d.ts

```diff
@@ -110,6 +110,8 @@ declare global {
       Fonts: Artifacts.Font[];
       /** Information on poorly sized font usage and the text affected by it. */
       FontSize: Artifacts.FontSize;
+      /** The issues surfaced in the devtools Issues panel */
+      Issues: Artifacts.Issue[];
```

Comment:
ideally we can start with just what we need in a new artifact and then add to it as our needs grow. In this case I think we're only going to start with a mixed content audit so we don't need same site cookie issues?

Starting minimal keeps the artifacts smaller for when they need to be saved (e.g. a future issue we don't care about (yet) could add data URLs that are humongous, taking up disk space but never looked at) and we don't have to worry about code depending on them and changing our mind later (technically these won't be part of our public artifacts yet but it could happen :)

Rearranging a bit could help audits using `Issues`, as well. To get the subset they want, any audit using an `Issues` artifact of this form would have to do something like
```js
const mixedContentIssues = artifacts.Issues
  .filter(issue => issue.code === 'MixedContentIssue');

for (const issue of mixedContentIssues) {
  // Check because `mixedContentIssueDetails` is optional
  if (!issue.details.mixedContentIssueDetails) continue;
  // use issue.details.mixedContentIssueDetails...
}
```

Instead, another form the artifacts could take would be something like
```ts
Issues: {
  mixedContentIssues: Crdp.Audits.MixedContentIssueDetails[];
}
```
which moves the filtering to a single place in the gatherer (saving the audits from doing type juggling) and allows for easy extension of the type (just add another issue property) when there's another type of issue we want to write an audit for.

Reply:
@Beytoven this is the shape I was suggesting fyi

Reply:
@brendankenny  Yeah, Connor and I had discussed this earlier and decided to let the team discuss it on this PR. I like the format that you and Connor have suggested for this use case. Just want to call out the trade off here is having to update the gatherer and type-definition (and by extension tests and mock data) any time a new or existing audit needs to consume an issue that isn't already used by another audit.